# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/nsAHttpConnection.h
# Commit: 00bb17e2a6a7
# Full Hash: 00bb17e2a6a73f9c66e861adae426818ff4d8caa
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2023-09-21 21:20:56
# Regressor Bug: 1784261
# File Overlap Count: 1
# Description:
#   Bug 1784261 - Add telemetry to understand the reason of connection closure, r=necko-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D188132
# ==============================================================================

diff -r e98350675c5a -r 00bb17e2a6a7 netwerk/protocol/http/nsAHttpConnection.h
--- a/netwerk/protocol/http/nsAHttpConnection.h	Thu Sep 21 08:42:26 2023 +0000
+++ b/netwerk/protocol/http/nsAHttpConnection.h	Thu Sep 21 08:59:03 2023 +0000
@@ -173,6 +173,7 @@
   virtual nsITRRSkipReason::value TRRSkipReason() = 0;
   virtual bool GetEchConfigUsed() = 0;
   virtual PRIntervalTime LastWriteTime() = 0;
+  virtual void SetCloseReason(ConnectionCloseReason aReason) = 0;
 };
 
 NS_DEFINE_STATIC_IID_ACCESSOR(nsAHttpConnection, NS_AHTTPCONNECTION_IID)
@@ -277,6 +278,9 @@
   bool GetEchConfigUsed() override {                                         \
     return (!(fwdObject)) ? false : (fwdObject)->GetEchConfigUsed();         \
   }                                                                          \
+  void SetCloseReason(ConnectionCloseReason aReason) override {              \
+    if (fwdObject) (fwdObject)->SetCloseReason(aReason);                     \
+  }                                                                          \
   PRIntervalTime LastWriteTime() override;
 
 // ThrottleResponse deliberately ommited since we want different implementation